### PR TITLE
fix: popover styles refactoring

### PR DESCRIPTION
**Pull request by platon-rov**
## Related Issue

Closes ngx https://github.com/SAP/fundamental-ngx/issues/7706

Follow up pr in ngx https://github.com/SAP/fundamental-ngx/pull/7832

## Description

Core popover styles refactoring.

### Before:

image

### After:

image

### CODE REVIEW

File: src/styles/popover.scss [hunk 1]
```diff
@@ -236,14 +244,20 @@ $block: #{$fd-namespace}-popover;
...
+    position: relative;
+    width: 100%;
     white-space: nowrap;
     box-shadow: $fd-popover-box-shadow;
     z-index: $fd-popover-z-index;
     border-radius: $fd-popover-border-radius;
     background: $fd-popover-background-color;
     opacity: 1;
 
+    &.fd-inline-help__content {
```
Comment (InnaAtanasova) [inline src/styles/popover.scss:256]: this is not allowed in styles. It's breaking the self-contained rule, which means that you can't modify the properties of one component inside another component. In this case you can't modify Inline Help content inside the block code for Popover.

File: src/styles/popover.scss [hunk 2]
```diff
@@ -279,82 +293,151 @@ $block: #{$fd-namespace}-popover;
...
         &::after {
           left: 1px;
-          margin: -$fd-popover-arrow-width-half 0;
           border-width: $fd-popover-arrow-width-half $fd-popover-arrow-height $fd-popover-arrow-width-half 0;
           border-right-color: $fd-popover-background-color;
         }
       }
+
+      &-x--start {
+        left: $fd-popover-arrow-offset;
```
Comment (InnaAtanasova) [inline src/styles/popover.scss:360]: shouldn't this have RTL?

File: src/styles/popover.scss [hunk 3]
```diff
@@ -279,82 +293,151 @@ $block: #{$fd-namespace}-popover;
...
+      }
+
+      &-x--center {
+        left: 50%;
+        transform: translateX(-50%);
+      }
+
+      &-x--end {
+        right: $fd-popover-arrow-offset;
```
Comment (InnaAtanasova) [inline src/styles/popover.scss:369]: shouldn't this have RTL?

File: src/styles/popover.scss [hunk 4]
```diff
@@ -279,82 +293,151 @@ $block: #{$fd-namespace}-popover;
...
+        top: 50%;
+        transform: translateY(-50%);
+      }
+
+      &-y--bottom {
+        bottom: $fd-popover-arrow-offset;
+      }
+    }
+
+    .fd-list {
```
Comment (InnaAtanasova) [inline src/styles/popover.scss:386]: Please check my comment about the self-contained elements.

File: src/styles/popover.scss [hunk 5]
```diff
@@ -279,82 +293,151 @@ $block: #{$fd-namespace}-popover;
...
+          @include set-list-border-radius(top);
+        }
+
+        &:last-child {
+          @include set-list-border-radius(bottom);
+        }
+      }
+    }
+
+    .fd-nested-list {
```
Comment (InnaAtanasova) [inline src/styles/popover.scss:400]: Same as above.

File: src/styles/popover.scss [hunk 6]
```diff
@@ -279,82 +293,151 @@ $block: #{$fd-namespace}-popover;
...
+                @include set-list-border-radius(bottom, 0);
+              }
+            }
+          }
+        }
+      }
+    }
+
+    .fd-side-nav,
+    .fd-nested-list__item {
```
Comment (InnaAtanasova) [inline src/styles/popover.scss:431]: Same as above. If it's needed you can rely on modifier classes.

**Review by InnaAtanasova — CHANGES_REQUESTED** [review]

File: src/styles/popover.scss (hunk 2 above)
Comment (platon-rov) [reply src/styles/popover.scss:360]: No, it shouldn't. See the example.

image

File: src/styles/popover.scss (hunk 3 above)
Comment (platon-rov) [reply src/styles/popover.scss:369]: https://github.com/SAP/fundamental-styles/pull/3215#discussion_r820681683

File: src/styles/popover.scss (hunk 4 above)
Comment (platon-rov) [reply src/styles/popover.scss:386]: Agree. But i'm not sure about changing it.

```
// Case 1
<fd-popover>
   <fd-list class="fd-list__popover"> // class can be easily added, refactoring is easy, however it's still the breaking change
   
// Case 2
<ng-template #template>
  // this template is for the popover, but there no information about it, it's handled explicitly
  // it's very hard to refactor
  <fd-list> 
```

File: src/styles/popover.scss (hunk 5 above)
Comment (platon-rov) [reply src/styles/popover.scss:400]: https://github.com/SAP/fundamental-styles/pull/3215#discussion_r820692761

File: src/styles/popover.scss (hunk 2 above)
Comment (platon-rov) [reply src/styles/popover.scss:360]: Appropriate css classes will be generated when initializing popover, then `start` will be `end` in RTL and vice versa.

**Review by mikerodonnell89 — APPROVED** [review]
works great when compiled and paired with https://github.com/SAP/fundamental-ngx/pull/7832

nice work 👍

**Review by InnaAtanasova — APPROVED** [review]

**Review by InnaAtanasova — COMMENTED** [review]
I approved the code but there are some visual issues you need to check:
Screen Shot 2022-03-11 at 11 25 46 AM
Screen Shot 2022-03-11 at 11 26 09 AM

Please see https://www.chromatic.com/pullrequest?appId=5f85e3904b19d600221872af&number=3215&view=changes

**Pull request merged**